### DISCUSSION TITLE
CORS error Pimcore10 /w DataHub 1.0.9

### DISCUSSION THREAD

**Discussion by snmabaur** [pimcore/pimcore#10687] [Q&A / Support]
Does anyone experiences a CORS error with Pimcore 10 and DataHub 1.0.9?

I have exactly the same setup with a Pimcore 6 and a Pimcore 10 and a Vue.js frontend... I fetch the data from localhost:3000 (frontend) to localhost:8080 with a JS fetch function.

On Pimcore 6 I get the data without problem on the endpoint //localhost:8080/pimcore-graphql-webservices/demo?apikey=XXXX but on Pimcore10 I get a CORS error

`Access to fetch at 'http://localhost:8080/pimcore-graphql-webservices/hotspot?apikey=XXX' from origin 'http://localhost:3000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

I tried to set the CORS headers in the nginx conf file  but that also does not work. When I create a controller with a custom route then it works with the nginx headers but not for the datahub bundle

Could it be that the DataHub Bundle is not supporting Pimcore10? 

Thanks

**Comment by carlossmedina** [comment]
Hello @snmabaur ,
i hope you are doing well. I experienced the same problem here. My solution was this:

`$response->headers->set('Access-Control-Allow-Headers', 'Origin, Content-Type, X-Auth-Token, x-api-key');`
Since i am sending the x-api-key as header too .

I send the apikey in the url and as header. In the first case i think the preflight cant be autenticated and need the apikey in the url. In the second one the api request itself must be authenticated.
 
I am using vuejs and send the requests with axios like this:
My URL: https://<domain>/pimcore-graphql-webservices/prodanet?apikey=some-key

My Headers:
`let headers = {
        headers: { 
          'x-api-key': 'some-key',
          'Content-Type': 'application/json',
        },
      };`
It works for us.
